Server/mods/deathmatch/logic/CGame.cpp

```diff
@@ -4711,17 +4710,34 @@ void CGame::RegisterClientTriggeredEventUsage(CPlayer* pPlayer)
     if (!pPlayer || !pPlayer->IsPlayer() || pPlayer->IsBeingDeleted())
         return;
 
-    m_mapClientTriggeredEvents[pPlayer]++;
+    auto now = GetTickCount64_();
+
+    // If key/player doesn't exist in map, store time of entry
+    if (m_mapClientTriggeredEvents.find(pPlayer) == m_mapClientTriggeredEvents.end())
+        m_mapClientTriggeredEvents[pPlayer].m_llTicks = now;
+
+    // Only increment if we haven't reached the interval time already
+    if (now - m_mapClientTriggeredEvents[pPlayer].m_llTicks <= m_iClientTriggeredEventsIntervalMs)
+        m_mapClientTriggeredEvents[pPlayer].m_uiCounter++;
 }
 
 void CGame::ProcessClientTriggeredEventSpam()
 {
-    for (const auto& [player, count]: m_mapClientTriggeredEvents)
+    for (const auto& [player, data]: m_mapClientTriggeredEvents)
     {
-        if (player && player->IsPlayer() && !player->IsBeingDeleted() && count > m_iMaxClientTriggeredEventsPerInterval)
-            player->CallEvent("onPlayerTriggerEventThreshold", {});
-    }
+        if (player && player->IsPlayer() && !player->IsBeingDeleted())
+        {
+            if (GetTickCount64_() - data.m_llTicks >= m_iClientTriggeredEventsIntervalMs)
+            {
+                if (data.m_uiCounter > m_iMaxClientTriggeredEventsPerInterval)
+                    player->CallEvent("onPlayerTriggerEventThreshold", {});
 
-    m_mapClientTriggeredEvents.clear();
-    m_lClientTriggeredEventsLastCheck = GetTickCount64_();
+                m_mapClientTriggeredEvents.erase(player);
```

Comment:
And I just noticed that you have an undefined behaviour here. Because you cannot erase an element while iterating this way. You have to use iterators. See first erase method: https://en.cppreference.com/w/cpp/container/map/erase. When you need to erase an element you write: `it = map.erase(it)`.

Reply:
If there was c++20 I would recommend std::erase_if, but.. we are still using on C++17.